lib/Dialect/TritonGPU/Transforms/AccelerateAMDMatmul.cpp

```diff
@@ -245,6 +253,84 @@ class BlockedToMFMA : public mlir::RewritePattern {
   }
 };
 
+class BlockedToWMMA : public mlir::RewritePattern {
+public:
+  BlockedToWMMA(mlir::MLIRContext *context)
+      : mlir::RewritePattern(tt::DotOp::getOperationName(), 2, context) {}
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::Operation *op,
+                  mlir::PatternRewriter &rewriter) const override {
+  auto dotOp = cast<tt::DotOp>(op);
+
+  auto oldRetType = dotOp.getResult().getType().cast<RankedTensorType>();
+    if (!oldRetType.getEncoding() ||
+        !oldRetType.getEncoding().isa<ttg::BlockedEncodingAttr>())
+      return failure();
+
+    if (!supportWMMA(dotOp))
+      return failure();
+
+    auto CTALayout = ttg::getCTALayout(oldRetType.getEncoding());
+
+    // get WMMA encoding for the given number of warps
+    auto retShape = oldRetType.getShape();
+    auto mod = op->getParentOfType<mlir::ModuleOp>();
+    int numWarps = ttg::TritonGPUDialect::getNumWarps(mod);
+
+    // operands
+    Value a = dotOp.getA();
+    Value b = dotOp.getB();
+    auto oldAType = a.getType().cast<RankedTensorType>();
+    auto oldBType = b.getType().cast<RankedTensorType>();
+    auto ctx = oldAType.getContext();
+
+    ttg::WmmaEncodingAttr wmmaEnc;
+
+    int nonKDim = 16;
+    int kDim = 16;
+
+    auto warpsPerTile = warpsPerTileWMMA(dotOp, retShape, numWarps);
+    wmmaEnc = ttg::WmmaEncodingAttr::get(oldRetType.getContext(), warpsPerTile, CTALayout);
+    auto newRetType =
+        RankedTensorType::get(retShape, oldRetType.getElementType(), wmmaEnc);
+
+    // convert accumulator
+    auto oldAcc = dotOp.getOperand(2);
+    auto newAcc = rewriter.create<ttg::ConvertLayoutOp>(oldAcc.getLoc(),
+                                                        newRetType, oldAcc);
+    auto oldAOrder = oldAType.getEncoding()
+                         .cast<ttg::DotOperandEncodingAttr>()
+                         .getParent()
+                         .cast<ttg::BlockedEncodingAttr>()
+                         .getOrder();
+    auto oldBOrder = oldBType.getEncoding()
+                         .cast<ttg::DotOperandEncodingAttr>()
+                         .getParent()
+                         .cast<ttg::BlockedEncodingAttr>()
+                         .getOrder();
+
+    // kWidth is a number of consecutive elements per one instruction per one
+    // thread
+    auto kWidth = kDim / 2;
+
+    auto newAType = RankedTensorType::get(
+        oldAType.getShape(), oldAType.getElementType(),
+        ttg::DotOperandEncodingAttr::get(ctx, 0, wmmaEnc, kWidth));
+    auto newBType = RankedTensorType::get(
+        oldBType.getShape(), oldBType.getElementType(),
+        ttg::DotOperandEncodingAttr::get(ctx, 1, wmmaEnc, kWidth));
+    a = rewriter.create<ttg::ConvertLayoutOp>(a.getLoc(), newAType, a);
+    b = rewriter.create<ttg::ConvertLayoutOp>(b.getLoc(), newBType, b);
+    auto newDot = rewriter.create<tt::DotOp>(dotOp.getLoc(), newRetType, a, b,
+                                             newAcc, dotOp.getAllowTF32(),
```

Comment:
The dotOperandLayout for wmma is different from that of mfma.

Reply:
Yes, it is completely different.
While creating an operand type (AType for example https://github.com/ROCmSoftwarePlatform/triton/pull/423/files#diff-cbbb00df02bbf76f4e85d73e50cd2f27b8a1d8d8b4db306e26cebee62012118bR317), I put there wmmaEnc.
I plan to implement layout specifics in conversions from/to dot layout. Is it acceptable, WDYT?